docs/aws/audit/route53monitoring/rules/domain_expiry_30_days_remediation.mdx

```diff
@@ -1,6 +1,268 @@
+   ```
+
+By following these steps, you can effectively monitor and prevent Route 53 domain expiry using Python scripts.
+</Accordion>
+
+</AccordionGroup>
+</Tab>
+<Tab title='Cause'>
+### Check Cause
+<AccordionGroup>
+<Accordion title='Using Console' defaultOpen='true'>
+1. Log in to your AWS Management Console.
+2. Navigate to the Route 53 dashboard. You can do this by typing "Route 53" into the search bar at the top of the console and selecting it from the dropdown menu.
+3. Once you're in the Route 53 dashboard, click on "Registered domains" in the left-hand menu. This will bring up a list of all the domains you have registered with Route 53.
+4. Check the "Expiration date" column for each domain. If a domain is set to expire in the next 30 days, it will be listed here.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. Install and configure AWS CLI: Before you can run any AWS CLI commands, you need to install and configure the AWS CLI on your local machine. You can do this by following the instructions provided in the AWS CLI User Guide.
+
+2. List hosted zones: Use the following command to list all the hosted zones in your AWS account:
+
+   ```
+   aws route53 list-hosted-zones
+   ```
+   This command will return a list of all the hosted zones in your AWS account, including their ID, name, and other information.
+
+3. List resource record sets: For each hosted zone, use the following command to list all the resource record sets:
+
+   ```
+   aws route53 list-resource-record-sets --hosted-zone-id <hosted-zone-id>
+   ```
+   Replace `<hosted-zone-id>` with the ID of the hosted zone you want to check. This command will return a list of all the resource record sets in the specified hosted zone, including their name, type, and other information.
+
+4. Check domain expiry: For each resource record set of type `SOA` (Start of Authority), check the `RDATA` field for the expiry value. If the expiry value is less than or equal to 30 days, then the domain is set to expire in 30 days or less. Note that the expiry value is specified in seconds, so you need to convert it to days by dividing by 86400 (the number of seconds in a day).
+
+   Unfortunately, AWS CLI does not provide a direct way to check the expiry date of a domain. You may need to use other tools or services, or check the domain expiry manually through the AWS Management Console.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install the necessary AWS SDK for Python (Boto3) if you haven't done so already. You can install it using pip:
+```python
+pip install boto3
+```
+
+2. Import the necessary libraries and initialize the Route53 client. You will need to provide your AWS credentials (access key and secret key) which you can get from your AWS console.
+```python
+import boto3
+from datetime import datetime, timedelta
+
+# Initialize the Route53 client
+route53 = boto3.client('route53', aws_access_key_id='YOUR_ACCESS_KEY', aws_secret_access_key='YOUR_SECRET_KEY')
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>